Zend/zend_builtin_functions.c

```diff
@@ -655,6 +647,8 @@ ZEND_FUNCTION(get_parent_class)
 		ce = Z_OBJ_P(arg)->ce;
 	} else if (Z_TYPE_P(arg) == IS_STRING) {
 		ce = zend_lookup_class(Z_STR_P(arg));
+	} else {
+		zend_argument_type_error(1, "must be of type object|string|null, %s given", zend_zval_type_name(arg));
```

Comment:
I'm wondering if we should rather try to cast the value to a string instead of requiring a string argument explicitly. This way we could leverage the usual type coercion semantics in weak mode, rather than always using the strict semantics.

P.S. If I'm not mistaken, I think this would be nice to fix for other functions too

Reply:
If there are multiple cases for String|Object you may want to create a dedicated Fast ZPP macro, that's what I did for int|string, we don't have a nice way to handle union types with ZPP currently so that's the best we can do ATM.

Reply:
Yes, that was my plan, too :)

Reply:
Oops, PEAR dies with `Uncaught TypeError: is_a(): Argument #1 ($object) must be of type object|string, array given in phar:///home/travis/build/php/php-src/pear/install-pear-nozlib.phar/PEAR.php:298` :D

Reply:
The check is too sloppy in https://github.com/pear/pear-core-minimal/blob/master/src/PEAR.php#L298